eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentHop.scala

```diff
@@ -0,0 +1,58 @@
+package fr.acinq.eclair.payment
+
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.wire.ChannelUpdate
+import fr.acinq.bitcoin.Protocol
+import java.nio.ByteOrder
+
+
+object PaymentHop {
+  /**
+    *
+    * @param baseMsat     fixed fee
+    * @param proportional proportional fee
+    * @param msat         amount in millisatoshi
+    * @return the fee (in msat) that a node should be paid to forward an HTLC of 'amount' millisatoshis
+    */
+  def nodeFee(baseMsat: Long, proportional: Long, msat: Long): Long = baseMsat + (proportional * msat) / 1000000
+
+  /**
+    *
+    * @param reversePath sequence of Hops from recipient to a start of assisted path
+    * @param msat an amount to send to a payment recipient
+    * @return a sequence of extra hops with a pre-calculated fee for a given msat amount
+    */
+  def buildExtra(reversePath: Seq[Hop], msat: Long): Seq[ExtraHop] = (List.empty[ExtraHop] /: reversePath) {
+    case (Nil, hop) => ExtraHop(hop.nodeId, hop.shortChannelId, hop.nextFee(msat), hop.cltvExpiryDelta) :: Nil
+    case (head :: rest, hop) => ExtraHop(hop.nodeId, hop.shortChannelId, hop.nextFee(msat + head.fee), hop.cltvExpiryDelta) :: head :: rest
+  }
+}
+
+trait PaymentHop {
+  def nextFee(msat: Long): Long
+  def shortChannelId: Long
+  def cltvExpiryDelta: Int
+  def nodeId: PublicKey
+}
+
+/**
+  * Extra hop contained in RoutingInfoTag
+  *
+  * @param nodeId          node id
+  * @param shortChannelId  channel id
+  * @param fee             node fee
+  * @param cltvExpiryDelta node cltv expiry delta
+  */
+case class ExtraHop(nodeId: PublicKey, shortChannelId: Long, fee: Long, cltvExpiryDelta: Int) extends PaymentHop {
```

Comment:
I'm not a fan of moving from PaymentRequest.scala to here, because it is inconsistent with other payment request-related classes, and contains serialization stuff.

Was there another reason apart from regrouping classes implementing the same trait?

Reply:
No other reasons. Should I put it back to `PaymentRequest.scala`?

Reply:
Yes, I'd rather keep all BOLT11 classes together